src/renderer/wg_engine/tvgWgCommon.cpp

```diff
@@ -327,6 +349,27 @@ void WgContext::releaseIndexBuffer(WGPUBuffer& buffer)
 }
 
 
+WGPUSurface WgContext::createSurface(WGPUInstance instance, void* disp_inst, void* wind_serf)
```

Comment:
I think we have two approaches to consider here:
1. Receive the generated instances of `WGPUSurface` / `WGPUInstance` from the user side instead of the window/display handles.
2. Add an additional parameter for the window type in the API so that we can properly identify the current window system.

So far, I think option 1 is better for ThorVG as it should not depend on the window system. However, we can improve this point further after this PR.

Reply:
1. So it will create dependence from webgpu.h header on the user side, where WGPUSurface/WGPUInstance types are described
2. Yes, its can be implemented. So we need to add public enum for a platform description

Reply:
> 1. So it will create dependence from webgpu.h header on the user side, where WGPUSurface/WGPUInstance types are described

So far, that's correct. The best approach is for the toolkit layer to abstract this point so apps don't depend on WebGPU directly.